src/AuthPM.js

```diff
@@ -1,6 +1,86 @@
+import AuthAPI from './AuthAPI';
+
 class AuthPM {
-  constructor() {
-    console.log('init');
+  #appId;
+
+  #target;
+
+  constructor(appId, target) {
+    this.#appId = appId;
+
+    this.#setTarget(target);
+
+    this.#target && this.#getAppList();
+  }
+
+  #setTarget(target) {
+    switch (typeof target) {
+      case 'object': {
+        if (AuthPM.#isDomElementExist(target)) {
+          this.#target = target;
+        }
+        break;
+      }
+      case 'string': {
+        const possibleTarget = document.querySelector(target);
+        if (AuthPM.#isDomElementExist(possibleTarget)) {
+          this.#target = possibleTarget;
+        }
+        break;
+      }
+      default: {
+        console.warn('Invalid type of target, check configuration');
+        return;
+      }
+    }
+
+    if (!this.#target) {
+      console.warn('Can not find target, check configuration');
+    }
+  }
+
+  #getAppList() {
+    AuthAPI.getAppList(this.#appId)
+      .then(({ socials }) => this.#renderSocials(socials))
+      .catch(() => {
+        console.warn(`Can not find application settings`);
+      });
+  }
+
+  #renderSocials(socials) {
+    socials.forEach(({ id, name }) => {
+      const btn = this.#renderButton({ id, name });
+
+      this.#target.appendChild(btn);
+    });
+  }
+
+  #renderButton({ id, name }) {
+    const clickHandler = (event) => {
```

Comment:
We need to save the handler to remove it on unmount.

Reply:
I think that this logic should be left to the user of the widget